src/WebJobs.Script/Description/Node/functions/functions.js

```diff
@@ -65,7 +69,12 @@ function createFunction(f) {
         inputs.unshift(context);
         delete context._inputs;
 
-        f.apply(null, inputs);
+        var userReturn = f.apply(null, inputs);
+        if (userReturn && typeof userReturn.then === 'function') {
+            context._promise = true;
+            userReturn.then((result) => context.done(null, result))
```

Comment:
@mathewc Added the bindings test, see: https://github.com/Azure/azure-webjobs-sdk-script/pull/673/files#diff-c79b5147f12f44c2f0ca539cbdb3ac21R6